adoptium/mirror-scripts
# feat: add logic to auto trigger release-pipeline once "-ga" tag arrive

Fix: https://github.com/adoptium/temurin-build/issues/3167

Test run: 
1. mock in 2022 Oct release when we have _adopt tag ready for jdk19 https://ci.adoptopenjdk.net/view/work-in-progress/job/wen_releaseTrigger/80/console
`Triggering projects: [test-release-openjdk19-pipeline](https://ci.adoptopenjdk.net/job/test-release-openjdk19-pipeline/)`
3. mock in 2022 Oct release we already built jdk19 from (1) no more need trigger 
```
Release tag jdk-19.0.1+10_adopt has been used in the current release
Will not continue job
Not triggering due to missing file
```

Detail:
1.  remove non-git tracked files: e.g properties from previous run
2.  either `git clone jdkXXu` or `git fetch` if the repo already there
3.  get expectedTag from r`eleasePlan.cfg` file
4.  check `workspace/tracking` file if exists from previous run and get the previous tag name
5.  compare the previous tag with expectedTag => if previous tag is used already, no need continue job, wont generate `properties` file => next step wont trigger due to lack of `preperties` file
6.  otherwise, continue to get the latest GA tag in the repo
7.  compare the GA tag and expectedTag to decide if GA tag is what we want for this release, if not wont continue job, wont generate `properties` file => next step wont trigger due to lack of `preperties` file
8. otherwise, try to find its _adopt tag on the same commit
9.  if `_adopt `tag exists (meaning, mirror job already applied tag)  , write into `preperties` file and exit to next step (trigger)
10. if `_adopt` tag not exist yet (mirror job still working/ongoing) , sleep `10mins `and loop` 4` times. 



Example:
created job for jdk8
https://ci.adoptopenjdk.net/job/build-scripts/job/utils/job/releaseTrigger_jdk8u/
which uses this PR's script and on scheduler of `0 H 17-20 1,3,4,7,9,10 1-5` refer to https://www.java.com/releases/



This should not auto trigger if it is jdk8 arm32 build: 
Ref: https://github.com/adoptium/ci-jenkins-pipelines/pull/525 to remove jdk8 arm32 linux from default settings.
